python/tank/util/metrics.py

```diff
@@ -324,16 +327,95 @@ def _dispatch(self, metrics):
+class EventMetric(object):
+    """
+    Convenience class for creating a metric event to be logged on a Shotgun site.
+
+    Use this helper class to create a suitable metric structure that you can
+    then pass to the `tank.utils.metrics.log_metric_event` method.
+
+    The simplest usage of this class is simply to provide an event group and
+    event name to the constructor:
+
+    Optionally, you can add your own specific metrics by using the
+    `properties` parameter. The latter simply takes a standard
+    dictionary.
+
+    The class also define numerous standard definition.
+    We highly recommand usage of them. Below is a complete typical usage:
+
+    ```
+    metric = EventMetric.log(EventMetric.GROUP_APP,
+        "User Logged In",
+        properties={
+            EventMetric.KEY_ENGINE: "tk-maya",
+            EventMetric.KEY_ENGINE_VERSION: "v0.2.2",
+            EventMetric.KEY_HOST_APP: "Maya",
+            EventMetric.KEY_HOST_APP_VERSION: "2017",
+            EventMetric.KEY_APP: "tk-multi-publish2",
+            EventMetric.KEY_APP_VERSION: "v0.2.3",
+            "CustomBoolMetric": True,
+            "RenderJobsSumitted": 173,
+        }
+    )
+    ```
+    """
 
-    def __init__(self, data):
-        """Initialize the object with a dictionary of metric data.
-        
-        :param dict data: A dictionary of metric data.
-        
+    # Event groups
+    GROUP_APP = "App"
+    GROUP_TASKS = "Tasks"
+    GROUP_MEDIA = "Media"
+    GROUP_TOOLKIT = "Toolkit"
+    GROUP_NAVIGATION = "Navigation"
+    GROUP_PROJECTS = "Projects"
+
+    # Event property keys
+    KEY_ACTION_TITLE = "Action Title"
+    KEY_APP = "App"
+    KEY_APP_VERSION = "App Version"
+    KEY_COMMAND = "Command"
+    KEY_ENGINE = "Engine"
+    KEY_ENGINE_VERSION = "Engine Version"
+    KEY_ENTITY_TYPE = "Entity Type"
+    KEY_HOST_APP = "Host App"
+    KEY_HOST_APP_VERSION = "Host App Version"
+    KEY_PUBLISH_TYPE = "Publish Type"
+
+    def __init__(self, group, name, properties=None):
+        """
+        Initialize a metric event using the specified parameters. Use this helper class
+        to create a suitable metric structure to be used with the
+
+        :param str group: A group or category this metric event falls into.
+        Although any values can be used, we encourage usage of the GROUP_*
+        definitions above.
+
+        :param: str name: A short descriptive event name or performed action.
+            The complete list can be found in the 'Shotgun Event Taxonomy' document.
+            Below are a few examples:
+                'Viewed Login Page'
+                'Logged In'
+                'Created Project'
+                'Toggled Project Favorite'
+                'Edited Task Status'
+                'Read Inbox Item'
+
+        :param: dict properties: An optional dictionary of extra properties to be attached to the metric event.
         """
-        self._data = data
+
+        """
```

Comment:
Should this be part of the docstring above? If it's just comments, then use `#`.